ldap/servers/slapd/back-ldbm/db-mdb/mdb_config.c

```diff
@@ -0,0 +1,1375 @@
+                "nsslapd-mdb-max-readers value is not compatible with current configuration."
+                " Increasing the value from %d to %d\n", val, curval);
+        val = curval;
+    }
+
+    if (apply) {
+        conf->dsecfg.max_readers = val;
+        if (CONFIG_PHASE_RUNNING == phase) {
+            slapi_log_err(SLAPI_LOG_NOTICE, "dbmdb_ctx_t_db_max_dbs_set",
+                "New nsslapd-mdb-max-dbs will not take affect until the server is restarted\n");
+        }
+    }
+
+    return retval;
+}
+
+static void *
+dbmdb_ctx_t_db_max_dbs_get(void *arg)
+{
+    struct ldbminfo *li = (struct ldbminfo *)arg;
+    dbmdb_ctx_t *conf = li->li_dblayer_config;
+
+    return  (void *)((uintptr_t)(conf->dsecfg.max_dbs));
+}
+
+static int
+dbmdb_ctx_t_db_max_dbs_set(void *arg, void *value, char *errorbuf __attribute__((unused)), int phase __attribute__((unused)), int apply)
+{
+    struct ldbminfo *li = (struct ldbminfo *)arg;
+    dbmdb_ctx_t *conf = li->li_dblayer_config;
+    int retval = LDAP_SUCCESS;
+    int val = (int)((uintptr_t)value);
+    int curval = val;
+
+    if (curval < conf->limits.min_dbs) {
+        curval = conf->limits.min_dbs;
+    }
+    if (val && curval != val) {
+        slapi_log_err(SLAPI_LOG_WARNING, "dbmdb_ctx_t_db_max_dbs_set",
+                "nsslapd-mdb-max-dbs value is not compatible with current configuration."
+                " Increasing the value from %d to %d\n", val, curval);
+        val = curval;
+    }
+
+    if (apply) {
+        conf->dsecfg.max_dbs = val;
+        if (CONFIG_PHASE_RUNNING == phase) {
+            slapi_log_err(SLAPI_LOG_NOTICE, "dbmdb_ctx_t_db_max_dbs_set",
+                "New nsslapd-mdb-max-dbs will not take affect until the server is restarted\n");
+        }
+    }
+
+    return retval;
+}
+static void *
+dbmdb_ctx_t_maxpassbeforemerge_get(void *arg)
+{
+    struct ldbminfo *li = (struct ldbminfo *)arg;
+
+    return (void *)((uintptr_t)(li->li_maxpassbeforemerge));
+}
+
+static int
+dbmdb_ctx_t_maxpassbeforemerge_set(void *arg, void *value, char *errorbuf __attribute__((unused)), int phase __attribute__((unused)), int apply)
+{
+    struct ldbminfo *li = (struct ldbminfo *)arg;
+    int retval = LDAP_SUCCESS;
+    int val = (int)((uintptr_t)value);
+
+    if (val < 0) {
+        slapi_log_err(SLAPI_LOG_NOTICE, "dbmdb_ctx_t_maxpassbeforemerge_set",
+                      "maxpassbeforemerge will not take negative value - setting to 100\n");
+        val = 100;
+    }
+
+    if (apply) {
+        li->li_maxpassbeforemerge = val;
+    }
+
+    return retval;
+}
+
+static void *
+dbmdb_ctx_t_db_durable_transactions_get(void *arg)
+{
+    struct ldbminfo *li = (struct ldbminfo *)arg;
+
+    return (void *)((uintptr_t)(MDB_CONFIG(li)->dsecfg.durable_transactions));
+}
+
+static int
+dbmdb_ctx_t_db_durable_transactions_set(void *arg, void *value, char *errorbuf __attribute__((unused)), int phase __attribute__((unused)), int apply)
```

Comment:
IMO we shouldn't enable or allow unsafe options like this ....

Reply:
If I remember rightly this one is about insuring that changes are really flushed on disk - and protect only against system crash and power failure. (IMHO the real risk is very limited (especially if power is secured and in multi supplier topology) and some people may want to assume it)
That said there are a lot of other options which are really unsafe, and like you I feel that we should not allow their use,

Reply:
We shouldn't allow a setting where people can turn this off is my point :) I think even with multiple powersources, redudant hardware, you have to consider memory corruption, or kernel panic, or a crash. This could cause dataloss and issues, so it's better to the correct thing over the fast thing. I don't really like handing dangerous weapons out in a project that prides itself on correctness :)

Reply:
Well I am of two mind about that one and will check what the other are thinking.
I understand very well your point of view and I agree with your arguments but on the other side 
 - the risks you mention exist with bdb since at least 20 years and as far as I know we never get bothered with those.
And if the feature got developed that was because it was needed by some customers. 
IMHO it has some interest:
  start the server with durable txn off to perform massive ldif imports 
  then restart the server after enabling durable txn on to be safe.

Reply:
Well, if that's the feature then it should be exposed as "non_durable_txn_during_import" rather than relying on the human operator to correctly "undo" the setting.

Just because it was something that has existed for 20 years doesn't mean we need to bring it forward now. And just because we haven't been bothered, doesn't mean someone has not been affected by it - we just were not told they were.

Reply:
just realizing I never wrote the conclusion of that point that was discussed some time ago in a meeting:
We will keep the non_durable_txn_during_import" (at least for now)